### DISCUSSION TITLE
SetAutoCalcFields for loops

### DISCUSSION THREAD

**Discussion by ChristianHovenbitzer** [StefanMaron/BusinessCentral.LinterCop#74] [New Rule]
How about a warning if a CalcFields is used inside a loop? In this szenario a SetAutoCalcFields should be be used before the FindSet.

Example
![image](https://user-images.githubusercontent.com/91181840/139851484-a93f21d0-4062-4003-8c33-0ad5b7cecabf.png)

**Comment by StefanMaron** [comment]
Are there any known exceptions for this? Where you would actually want to do a calcfields within the loop?

**Reply by ChristianHovenbitzer** [reply]
I can't think of any exception for this. Perhaps if there is a condition within the loop but I am not sure if this would even be a performance improvment. Considering readability I would still go for an AutoCalcFields
